crates/evm-prover/src/prover/prover.rs

```diff
@@ -100,29 +50,41 @@ pub struct AppContext {
     pub namespace: Namespace,
     pub celestia_rpc: String,
     pub evm_rpc: String,
-    pub sequencer_rpc: String,
+    pub pub_key: Vec<u8>,
+    pub trusted_state: RwLock<TrustedState>,
+}
+
+/// TrustedState tracks the current height and state root which is provided to the proof system as its trusted inputs.
+pub struct TrustedState {
+    height: u64,
+    root: FixedBytes<32>,
 }
 
 impl AppContext {
     pub fn from_config(config: Config) -> Result<Self> {
         let genesis = AppContext::load_genesis().context("Error loading app genesis")?;
-
-        let chain_spec = Arc::new(
+        let chain_spec: Arc<ChainSpec> = Arc::new(
             (&genesis)
                 .try_into()
                 .map_err(|e| anyhow!("Failed to convert genesis to chain spec: {e}"))?,
         );
 
         let raw_ns = hex::decode(config.namespace_hex)?;
         let namespace = Namespace::new_v0(raw_ns.as_ref()).context("Failed to construct Namespace")?;
+        let pub_key = hex::decode(config.pub_key)?;
+        let trusted_state = RwLock::new(TrustedState {
+            height: 0,
+            root: chain_spec.genesis_hash(),
```

Comment:
The prover service currently starts with the evm app's genesis hash. 
Alternatively, we can query the latest from the node on startup and use that + its height instead. 

Any thoughts?

Reply:
I think ideal would be to query the reth node on startup for height 0, but seems like a low priority thing to do

Reply:
Agreed, it is low priority for now. Querying for height 0 would yield the same state root available in the genesis. 
And I just realised as I wrote this response that I actually put the wrong value into this field, as genesis hash != genesis state root.

What I want here is `chain_spec.genesis_header().state_root`, I will update! 🙏🏻